emily_cron/app/services/deposit_processor.py

```diff
@@ -0,0 +1,191 @@
+            if spending_tx_details:
+                for vin in spending_tx_details.get("vin", []):
+                    if (
+                        vin.get("txid") == tx.bitcoin_txid
+                        and vin.get("vout") == tx.bitcoin_tx_output_index
+                    ):
+                        witness_data = vin.get("witness", [])
+                        # Simple check: does witness contain the reclaim script?
+                        if any(tx.reclaim_script in item for item in witness_data):
+                            is_reclaim_check = True
+                            break
+            else:
+                logger.warning(
+                    f"Could not fetch spending tx details for {spending_txid} to check for reclaim of {tx.bitcoin_txid}."
+                )
+                # If we can't fetch the spending tx, we can't confirm reclaim.
+                # For safety, assume it might be a signer sweep.
+                continue
+
+            if is_reclaim_check:
+                # Case 2: Time expired, UTXO spent, AND identified as reclaim -> Mark FAILED
+                logger.info(
+                    f"Marking transaction {tx.bitcoin_txid} as FAILED (Depositor reclaim detected)"
+                )
+                updates.append(
+                    DepositUpdate(
+                        bitcoin_txid=tx.bitcoin_txid,
+                        bitcoin_tx_output_index=tx.bitcoin_tx_output_index,
+                        status=RequestStatus.FAILED.value,
+                        status_message=f"Depositor reclaim detected in tx {spending_txid}",
+                    )
+                )
+            else:
+                # Case 3: Time expired, UTXO spent, but NOT identified as reclaim -> Assume signer sweep, do nothing.
+                logger.debug(
+                    f"Deposit {tx.bitcoin_txid} time expired, but UTXO spent by {spending_txid} (likely signer sweep). Not failing."
+                )
+
+        if updates:
+            logger.info(
+                f"Found {len(updates)} transactions to mark as FAILED (expired or reclaimed)"
+            )
+
+        return updates
+
+    def update_deposits(self) -> None:
+        """Update deposit statuses.
+
+        This is the main entry point for the deposit processor.
+        It fetches deposits, enriches them, processes them, and updates their status.
+        """
+        logger.info("Running deposit status update job")
+
+        # Get current blockchain state
+        bitcoin_chaintip_height = MempoolAPI.get_tip_height()
+        stacks_chaintip = HiroAPI.get_stacks_block()
+
+        logger.info(f"Bitcoin chain tip: {bitcoin_chaintip_height}")
+        logger.info(f"Stacks chain tip: {stacks_chaintip}")
+
+        # Fetch pending and accepted deposits
+        pending_deposits = PrivateEmilyAPI.fetch_deposits(RequestStatus.PENDING)
+        accepted_deposits = PrivateEmilyAPI.fetch_deposits(RequestStatus.ACCEPTED)
```

Comment:
nit: can we add a comment explaining why we are also checking the accepted ones? IIUC it's because emily cannot detect bitcoin only forks, so we could have accepted deposits forked (and invalidated) but still in an accepted state